var/spack/repos/builtin/packages/hpcc/package.py

```diff
@@ -0,0 +1,207 @@
+            for k, v in self.config.items():
+                make_arch.filter(k, v)
+
+    def edit(self, spec, prefix):
+        # Message Passing library (MPI)
+        self.config['@MPINC@'] = spec['mpi'].headers.include_flags
+        self.config['@MPLIB@'] = spec['mpi'].libs.search_flags
+
+        lin_alg_libs = []
+        # FFT
+        if self.spec.variants['fft'].value in ('fftw2', 'mkl'):
+            self.config['@LAINC@'] += ' -DUSING_FFTW'
+
+            if self.spec.variants['fft'].value == 'fftw2':
+                self.config['@LAINC@'] += \
+                    spec['fftw-api'].headers.include_flags
+                # fftw does not set up libs for version 2
+                lin_alg_libs.append(
+                    join_path(spec['fftw-api'].prefix.lib, 'libsfftw_mpi.so'))
+                lin_alg_libs.append(
+                    join_path(spec['fftw-api'].prefix.lib, 'libsfftw.so'))
+
+            elif self.spec.variants['fft'].value == 'mkl' and '^mkl' in spec:
+                mklroot = env['MKLROOT']
+                self.config['@LAINC@'] += \
+                    ' -I{0}'.format(join_path(mklroot, 'include/fftw'))
+                libfftw2x_cdft = join_path(
+                    mklroot, 'lib', 'intel64', 'libfftw2x_cdft_DOUBLE_ilp64.a')
+                libfftw2xc = join_path(
+                    mklroot, 'lib', 'intel64', 'libfftw2xc_double_intel.a')
+                if not (os.path.exists(libfftw2x_cdft) and
+                        os.path.exists(libfftw2xc)):
+                    raise InstallError(
+                        "HPCC need fftw2 interface, "
+                        "here are brief notes how to make one:\n"
+                        "# make fftw2x_cdft interface to mkl\n"
+                        "cd $MKLROOT/interfaces/fftw2x_cdft\n"
+                        "make libintel64 PRECISION=MKL_DOUBLE "
+                        "interface=ilp64 MKLROOT=$MKLROOT\n"
+                        "\n"
+                        "# make FFTW C wrapper library\n"
+                        "cd $MKLROOT/interfaces/fftw2xc\n"
+                        "make libintel64 PRECISION=MKL_DOUBLE "
+                        "MKLROOT=$MKLROOT\n")
+                lin_alg_libs.append(libfftw2xc)
+                lin_alg_libs.append(libfftw2x_cdft)
+
+        # Linear Algebra library (BLAS or VSIPL)
+        self.config['@LAINC@'] = spec['blas'].headers.include_flags
+        lin_alg_libs = lin_alg_libs + [
+            lib for lib in spec['blas'].libs if lib not in lin_alg_libs]
+
+        # pack all LA/FFT libraries
+        self.config['@LALIB@'] = ' '.join(lin_alg_libs)
+
+        # Compilers / linkers - Optimization flags
+        self.config['@CC@'] = '{0}'.format(spec['mpi'].mpicc)
+
+        # Compiler flags for CPU architecture optimizations
+        if spec.satisfies('%intel') and \
+                (spec.satisfies('^intel-parallel-studio+mpi') or
+                 spec.satisfies('^intel-mpi')):
```

Comment:
I think this should only be `spec.satisfies('%intel')`. A user could theoretically be building with `intel-mpi` or `intel-parallel-studio+mpi` but with the GCC compilers. The compiler flags are only relevant to the Intel compiler, right?

Reply:
That is a very good suggestion. I am just not very sure how spack's compilers wrappers works. Using extra flags to gcc (-Q --help=target) and bash wrapper around icc, I found that -march/-mtune flags is passed in case of gcc/open-mpi but not in case of icc/intel-mpi. I didn't test gcc/intel-mpi combination and don't recall icc/openmpi outcome. 

Some other packages still pass -march/-mtune flags even in case of gcc (for example namd). So may be it have sense to remove the whole condition and always pass -march/-mtune flags. 

What do you think about that?

Reply:
The `spec.target.optimization_flags` are passed via the compiler wrappers whether or not it is done manually, so you might not need the whole thing.

Reply:
By whole thing, do you mean 
1) the condition with body, that is remove that:
```python
        if spec.satisfies('%intel') and \
                (spec.satisfies('^intel-parallel-studio+mpi') or
                 spec.satisfies('^intel-mpi')):
            # with intel-parallel-studio+mpi the '-march' arguments
            # are not passed to icc
            arch_opt = spec.target.optimization_flags(
                spec.compiler.name, spec.compiler.version)
            self.config['@CCFLAGS@'] = '-O3 {0}'.format(arch_opt)
```
2) or only conditions, that is remove condition and keep this:
```python
            arch_opt = spec.target.optimization_flags(
                spec.compiler.name, spec.compiler.version)
            self.config['@CCFLAGS@'] = '-O3 {0}'.format(arch_opt)
```

In the first case I am not convinced that march.mtune flags is passed in case of intel compilers.

Thanks,
Nikolay

Reply:
I meant 1. As long as the compiler wrappers are used, the flags will be injected.

Reply:
Right, but it looks like they either not used or not used properly in case on intel compilers at this point. I made a simple wrapper around icc which just prints arguments and pass them to icc, i didn't see any architecture flag.

I went with your initial suggestion and removes `(spec.satisfies('^intel-parallel-studio+mpi') or spec.satisfies('^intel-mpi'))` from condition. Plus I need some extra flags in case of not extremely new icc versions as well.